test/ambassador_test.go

```diff
@@ -0,0 +1,153 @@
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	ambassadorv2 "github.com/datawire/ambassador/pkg/api/getambassador.io/v2"
+	testcluster "github.com/mesosphere/ksphere-testing-framework/pkg/cluster"
+	testharness "github.com/mesosphere/ksphere-testing-framework/pkg/harness"
+	"github.com/mesosphere/kubeaddons/pkg/api/v1beta2"
+	"github.com/mesosphere/kubeaddons/pkg/constants"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+const waitForAmbassador = time.Minute * 1
+
+func ambassadorChecker(t *testing.T, cluster testcluster.Cluster) testharness.Job {
+	return func(t *testing.T) error {
+		app := "quote"
+		ns := "default"
+
+		// create a test application to ensure proper connectivity
+		testDeployment := appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      app,
+				Namespace: ns,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: pointer.Int32Ptr(1),
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": app}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": app},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{{
+							Name:  "backend",
+							Image: "docker.io/datawire/quote:0.4.1",
+							Ports: []corev1.ContainerPort{{
+								Name:          "http",
+								ContainerPort: 8080,
+							}}}}}}}}
+		deployment, err := cluster.Client().AppsV1().Deployments(ns).Create(context.TODO(), &testDeployment, metav1.CreateOptions{})
```

Comment:
Do we want to wait for deployment to finish before we continue?

Reply:
You're right that we probably do want to do this, I'll make an update for it.